custom/ewsghana/api.py

```diff
@@ -178,40 +178,99 @@ def _create_supply_point_from_location(self, supply_point, location):
-                         administrative=False)
-        ]
-        domain.save()
+        country, _ = LocationType.objects.get_or_create(
+            domain=self.domain,
+            name="country",
+            administrative=True,
+        )
+        LocationType.objects.get_or_create(
+            domain=self.domain,
+            name="Central Medical Store",
+            administrative=False,
+            parent_type=country,
+        )
+        LocationType.objects.get_or_create(
+            domain=self.domain,
+            name="Teaching Hospital",
+            administrative=False,
+            parent_type=country,
+        )
+
+        region, _ = LocationType.objects.get_or_create(
+            domain=self.domain,
+            name="region",
+            administrative=True,
+            parent_type=country,
+        )
+        LocationType.objects.get_or_create(
+            domain=self.domain,
+            name="Regional Medical Store",
+            administrative=False,
+            parent_type=region,
+        )
+        LocationType.objects.get_or_create(
+            domain=self.domain,
+            name="Regional Hospital",
+            administrative=False,
+            parent_type=region,
+        )
+
+        district, _ = LocationType.objects.get_or_create(
+            domain=self.domain,
+            name="district",
+            administrative=True,
+            parent_type=region,
+        )
+        LocationType.objects.get_or_create(
+            domain=self.domain,
+            name="Clinic",
+            administrative=False,
+            parent_type=district,
+        )
+        LocationType.objects.get_or_create(
+            domain=self.domain,
+            name="District Hospital",
+            administrative=False,
+            parent_type=district,
+        )
+        LocationType.objects.get_or_create(
+            domain=self.domain,
+            name="Health Centre",
+            administrative=False,
+            parent_type=district,
+        )
+
+        LocationType.objects.get_or_create(
+            domain=self.domain,
+            name="CHPS Facility",
+            administrative=False,
+            parent_type=district,
+        ),
+        LocationType.objects.get_or_create(
+            domain=self.domain,
+            name="Hospital",
+            administrative=False,
+            parent_type=district,
+        )
+        LocationType.objects.get_or_create(
+            domain=self.domain,
+            name="Psychiatric Hospital",
+            administrative=False,
+            parent_type=district,
+        )
+        LocationType.objects.get_or_create(
+            domain=self.domain,
+            name="Polyclinic",
+            administrative=False,
+            parent_type=district,
+        )
+        LocationType.objects.get_or_create(
+            domain=self.domain,
+            name="facility",
+            administrative=False,
+            parent_type=district,
+        )
```

Comment:
(not actionable) it might be more readable if you just made a datastructure to hold the whole structure. i realize that would be a little annoying because you'd have to make and map the parents first though.